api/src/main/scala/dcos/metronome/api/v1/controllers/JobRunController.scala

```diff
@@ -31,7 +31,7 @@ class JobRunController(
   def getJobRun(id: JobId, runId: String) = AuthorizedAction.async { implicit request =>
     async {
       await(jobRunService.getJobRun(JobRunId(id, runId))) match {
-        case Some(run) => request.authorized(ViewRunSpec, run, Ok(run))
+        case Some(run) => request.authorized(ViewRunSpec, run.jobRun.jobSpec, Ok(run))
```

Comment:
can we somehow use this method to not duplicate the logic to one more place? https://github.com/dcos/metronome/blob/4336d5b15cdd19be4088cd0d68268c5f9b96fc4d/api/src/main/scala/dcos/metronome/api/Authorization.scala#L39

Reply:
yeah.. I looked at doing that... it made the code hard to read.  I will look again.

Reply:
this issue is isAllowed is a predicate used to filter... authorized is a function that returns the block... there would aways be a function in this case similar to what the `authorized` is doing IMO

Reply:
and technically... `isAllowed` eventually calls `request.authorized` so it isn't redundant...  The issue I'm trying to understand but isn't blocking this merge is... `request.authorized` should take a matching `action: AuthorizedAction[Auth], resource: Auth`... the AuthorizedAction[Auth] is `ViewRunSpec` which is `AuthorizedAction[RunSpec]` resource should be required to be `RunSpec` which `JobSpec` is... however the compiler does NOT check this.    Leaving me confused